### PR TITLE
HodlHTLC Mode!

### PR DESCRIPTION
**Pull request by cfromknecht**
This commit adds a new debug mode for lnd called `hodlhtlc`. This mode instructs a node to refrain from settling incoming HTLCs for which it is the exit node. We plan to use this in testing to more precisely control the states a node can take during execution.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -113,6 +113,7 @@ type config struct {
 	RPCPort            int  `long:"rpcport" description:"The port for the rpc server"`
 	RESTPort           int  `long:"restport" description:"The port for the REST server"`
 	DebugHTLC          bool `long:"debughtlc" description:"Activate the debug htlc mode. With the debug HTLC mode, all payments sent use a pre-determined R-Hash. Additionally, all HTLCs sent to a node with the debug HTLC R-Hash are immediately settled in the next available state transition."`
+	HodlHTLC           bool `long:"hodlhtlc" description:"Activate the hodl HTLC mode.  With hodl HTLC mode, all incoming HTLCs will be accepted by the recieving node, but no attempt will be made to settle the payment with the sender."`
```
Comment (jimpo) [inline config.go]: typo: `receiving`

File: htlcswitch/link.go [hunk 2]
```diff
@@ -1153,6 +1159,13 @@ func (l *channelLink) processLockedInHtlcs(
 					continue
 				}
 
+				if l.cfg.DebugHTLC && l.cfg.HodlHTLC {
+					log.Errorf("hodl HTLC mode enabled, " +
```
Comment (jimpo) [inline htlcswitch/link.go]: Warn seems like a more appropriate log level to me.

File: config.go (hunk 1 above)
Comment (cfromknecht) [reply config.go]: Nice catch, thanks!

File: htlcswitch/link.go (hunk 2 above)
Comment (cfromknecht) [reply htlcswitch/link.go]: 100% agree 🎉

**Pull request merged**